addons/stock/views/stock_quant_views.xml

```diff
@@ -74,14 +89,28 @@
         <field name="model">stock.quant</field>
         <field eval="10" name="priority"/>
         <field name="arch" type="xml">
-            <tree string="Inventory Valuation" create="0" delete="0" decoration-danger="quantity &lt; 0">
-                <field name="product_id"/>
-                <field name="location_id"/>
-                <field name="lot_id" groups="stock.group_production_lot"/>
-                <field name="package_id" groups="stock.group_tracking_lot"/>
-                <field name="owner_id" groups="stock.group_tracking_owner"/>
-                <field name="reserved_quantity" string="Reserved"/>
-                <field name="quantity" string="On Hand"/>
+            <tree decoration-danger="quantity &lt; 0">
+                <field name="id" invisible="1"/>
+                <field name="tracking" invisible="1"/>
+                <field name="product_id" attrs="{'readonly': [('id', '!=', False)]}"
+                       readonly="context.get('single_product', False)" force_save="1"/>
+                <field name="location_id" attrs="{'readonly': [('id', '!=', False)]}"
+                       invisible="context.get('hide_location', False)"/>
+                <field name="lot_id" groups="stock.group_production_lot"
+                       attrs="{
+                            'readonly': ['|', ('id', '!=', False), ('tracking', 'not in', ['serial', 'lot'])],
+                            'required': [('tracking', '!=', 'none')]
+                       }"
+                       context="{'default_product_id': product_id}"
+                       domain="[('product_id', '=', product_id)]"/>
+                <field name="package_id" groups="stock.group_tracking_lot"
+                       attrs="{'readonly': [('id', '!=', False)]}"/>
+                <field name="owner_id" groups="stock.group_tracking_owner"
+                       attrs="{'readonly': [('id', '!=', False)]}"/>
+                <field name="reserved_quantity"/>
+                <field name="quantity" invisible="context.get('inventory_mode')"/>
+                <field name="inventory_quantity" string="On Hand Quantity"
+                       invisible="not context.get('inventory_mode')"/>
```

Comment:
these conditions are duplicated from the fields_view override and they don't take into account the "inventory_mode" context key (meaning the tree could be editable but the write will fail server side)

we need to do better here